src/printing_solver.cpp

```diff
@@ -40,5 +42,247 @@ namespace smt {
 /* PrintingSolver */
 
 // implementations
+PrintingSolver::PrintingSolver(SmtSolver s, std::ostream* os, PrintingStyleEnum pse)
+    : AbsSmtSolver(s->get_solver_enum()),
```

Comment:
What do you think. Should a printing solver get its own enum? I'm not sure if it needs one (i.e. I think it's fine how you've done it). But what do you think? Would the user expect a different solver enum for a CVC4Solver vs a PrintingSolver that wraps a CVC4Solver?

Reply:
yeah i wasn't sure about that either... My current code is a bit guilty in misrepresenting the actual state, but having more enums for printing solvers seems bloated... I'll leave the decision to you :)

Reply:
Yeah it would be a lot of enums. Let's leave it as-is for now! If it becomes important, we can change it later.